crates/stages/src/db.rs

```diff
@@ -164,6 +164,26 @@ where
         Ok((prev_body.start_tx_id + prev_body.tx_count, last_transition))
     }
 
+    /// Get the result vector with range [start_key:end_key).
+    /// Please note that start_key is inclusive and end_key is expclusive.
+    #[allow(clippy::type_complexity)]
```

Comment:
I've got the clippy warning, `very complex type used. Consider factoring parts into type definitions` for the return type, `Result<Vec<(T::Key, T::Value)>, Error>`.

If defining the type, `type VecPairResult<T> = Result<Vec<(<T as Table>::Key, <T as Table>::Value)>, Error>;`, I've get the error, `error[E0658]: inherent associated types are unstable`.

How to resolve it? Please could anyone give me an advice?